routes/index.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+var express = require('express');
+var router = express.Router();
+const mongoose = require('mongoose');
+const User = require('../modules/userdata');
+
+// BCrypt to encrypt passwords
+
+const bcrypt = require('bcrypt');
+const bcryptSalt = 10;
+
+// -- Connect to mongoose
+
+mongoose.Promise = Promise;
+mongoose.connect('mongodb://localhost/cookie-starter', {
+  keepAlive: true,
+  reconnectTries: Number.MAX_VALUE
+});
+
+/* GET home page. */
+
+router.get('/', (req, res, next) => {
+  res.render('auth/sign-up', { title: 'Express' });
+});
+
+router.get('/auth/sign-up', (req, res, next) => {
+  res.render('auth/sign-up', { title: 'Express' });
+});
+
+router.post('/new-user', (req, res, next) => {
+  const name = req.body.name;
+  const email = req.body.email;
+  const username = req.body.username;
+  const password = req.body.password;
+  const salt = bcrypt.genSaltSync(bcryptSalt);
+  const hashPass = bcrypt.hashSync(password, salt);
+
+  User.findOne({username: username})
+    .then(user => {
+      if (!user) {
+        return res.redirect('/auth/login');
+      } else if (user === username) {
+        return res.redirect('/auth/login');
+      } else {
+        const user = new User({
```

Comment:
Not sure if this is working as you intent. If you don't find any user here then you want to create it but in this case, when you don't find a user, you are redirecting to /auth/login